src/irc/ClientPool.ts

```diff
@@ -414,21 +415,16 @@ export class ClientPool {
         const cliConfig = bridgedClient.getClientConfig();
         cliConfig.setDesiredNick(bridgedClient.nick);
 
-        if (!bridgedClient.matrixUser) {
-            // no associated matrix user, run away!
-            return;
-        }
-
         const cli = this.createIrcClient(
-            cliConfig, bridgedClient.matrixUser, bridgedClient.isBot
+            cliConfig, bridgedClient.matrixUser || null, bridgedClient.isBot
         );
         const chanList = bridgedClient.chanList;
         // remove ref to the disconnected client so it can be GC'd. If we don't do this,
         // the timeout below holds it in a closure, preventing it from being GC'd.
         (bridgedClient as unknown) = undefined;
 
-        if (chanList.length === 0) {
-            log.info(`Dropping ${cli.id} ${cli.nick} because they are not joined to any channels`);
+        if (chanList.length === 0 && !bridgedClient.isBot) { // Never drop the bot.
+            log.info(`Dropping ${cli.id} (${cli.nick}) because they are not joined to any channels`);
```

Comment:
likewise

Reply:
Same deal. Not many people disconnect who have also never joined a channel. You'd expect to see both these messages possibly 50 times a day at the worst case on FN, which would put them in the info category for me.